x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeServiceSettings.java

```diff
@@ -46,19 +37,7 @@ public static ElserMlNodeServiceSettings.Builder fromMap(Map<String, Object> map
         Integer numAllocations = ServiceUtils.removeAsType(map, NUM_ALLOCATIONS, Integer.class);
         Integer numThreads = ServiceUtils.removeAsType(map, NUM_THREADS, Integer.class);
 
-        if (numAllocations == null) {
-            validationException.addValidationError(
-                ServiceUtils.missingSettingErrorMsg(NUM_ALLOCATIONS, ModelConfigurations.SERVICE_SETTINGS)
-            );
-        } else if (numAllocations < 1) {
-            validationException.addValidationError(mustBeAPositiveNumberError(NUM_ALLOCATIONS, numAllocations));
-        }
-
-        if (numThreads == null) {
-            validationException.addValidationError(ServiceUtils.missingSettingErrorMsg(NUM_THREADS, ModelConfigurations.SERVICE_SETTINGS));
-        } else if (numThreads < 1) {
-            validationException.addValidationError(mustBeAPositiveNumberError(NUM_THREADS, numThreads));
-        }
```

Comment:
moved to the below function (`validateParameters`) of MlNodeDeployedServiceSettings